### PR TITLE
atc: remove duplicate team roles

### LINKED ISSUE #5524: duplicated roles in `/api/v1/user` response
## Summary

it seems like we've got a regression in the `/api/v1/user` endpoint:

```
fly -t local userinfo --verbose

(...)

{"sub":"CgR0ZXN0EgVsb2NhbA","name":"","user_id":"test","user_name":"test","email":"test","is_admin":true,"is_system":false,"teams":{"main":["owner","owner"]}}

username  team/role
test      main/owner,main/owner
```

## Steps to reproduce

1. docker-compose up -d --build
2. fly -t local login -u test -p test
3. fly -t local userinfo --json

## Actual results

```
{
  "email": "test",
  "is_admin": true,
  "is_system": false,
  "name": "",
  "sub": "CgR0ZXN0EgVsb2NhbA",
  "teams": {
    "main": [
      "owner",
      "owner"
    ]
  },
  "user_id": "test",
  "user_name": "test"
}
```

## Expected results

```
{
  "email": "test",
  "is_admin": true,
  "is_system": false,
  "name": "",
  "sub": "CgR0ZXN0EgVsb2NhbA",
  "teams": {
    "main": [
      "owner",
     ]
  },
  "user_id": "test",
  "user_name": "test"
}
```

## Additional context

this was caught by k8s-topgun here:

https://github.com/concourse/concourse/blob/3b487b5b48a4ff6039330cd98b65c3f79b6eb80d/topgun/k8s/mainteam_role_test.go#L42-L60

## Triaging info

* Concourse version: (what will become 6.1.x - 421cf38c0d6df9576ff2e7d254f9f121c364a3af)
* Did this used to work? yes

### PR DESCRIPTION
**Pull request by pivotal-jwinters**
Don't add duplicate roles to /api/v1/user response

fixes #5524

### CODE REVIEW

File: atc/api/accessor/accessor.go [hunk 1]
```diff
@@ -135,18 +135,18 @@ func (a *access) rolesForTeam(auth atc.TeamAuth) []string {
 
 		// backwards compatibility for allow-all-users
 		if len(userAuth) == 0 && len(groupAuth) == 0 {
-			roles = append(roles, role)
+			roleSet[role] = true
```
Comment (cirocosta) [inline atc/api/accessor/accessor.go:138]: it seems like we're not testing this anywhere - should we have a test that covers it?

File: atc/api/accessor/accessor.go [hunk 2]
```diff
@@ -135,18 +135,18 @@ func (a *access) rolesForTeam(auth atc.TeamAuth) []string {
...
 			if userID != "" {
 				if strings.EqualFold(user, fmt.Sprintf("%v:%v", connectorID, userID)) {
-					roles = append(roles, role)
+					roleSet[role] = true
 				}
 			}
 			if userName != "" {
 				if strings.EqualFold(user, fmt.Sprintf("%v:%v", connectorID, userName)) {
-					roles = append(roles, role)
+					roleSet[role] = true
```
Comment (cirocosta) [inline atc/api/accessor/accessor.go:149]: I think this one is not tested either

File: atc/api/accessor/accessor.go (hunk 1 above)
Comment (pivotal-jwinters) [reply atc/api/accessor/accessor.go:138]: nah this is legacy

File: atc/api/accessor/accessor.go (hunk 2 above)
Comment (pivotal-jwinters) [reply atc/api/accessor/accessor.go:149]: yeah i think some of these tests got dropped when we moved this stuff around. re-added!

**Pull request merged**